swarm-test/src/lib.rs

```diff
@@ -230,7 +230,7 @@ where
 
         let dial_opts = DialOpts::peer_id(*other.local_peer_id())
             .addresses(external_addresses)
-            .condition(PeerCondition::Always)
+            .dial_conditions(DialConditions::empty())
```

Comment:
Should we maybe call this:

```suggestion
            .only_dial_if(DialConditions::empty())
```

It would be good to add an additional function `always` so that this reads better.